fineract-provider/src/main/java/org/apache/fineract/infrastructure/springbatch/messagehandler/kafka/KafkaWorkerConfig.java

```diff
@@ -0,0 +1,105 @@
+import org.apache.fineract.infrastructure.springbatch.messagehandler.conditions.kafka.KafkaWorkerCondition;
+import org.apache.kafka.common.serialization.IntegerDeserializer;
+import org.springframework.batch.integration.partition.StepExecutionRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.integration.channel.QueueChannel;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.kafka.config.ConcurrentKafkaListenerContainerFactory;
+import org.springframework.kafka.core.ConsumerFactory;
+import org.springframework.kafka.core.DefaultKafkaConsumerFactory;
+import org.springframework.kafka.listener.ContainerProperties;
+import org.springframework.kafka.support.Acknowledgment;
+import org.springframework.kafka.support.serializer.JsonDeserializer;
+import org.springframework.messaging.handler.annotation.Payload;
+
+@Configuration
+@Conditional(KafkaWorkerCondition.class)
+@Slf4j
+public class KafkaWorkerConfig {
+
+    @Autowired
+    private QueueChannel inboundRequests;
+
+    @Autowired
+    private StepExecutionRequestHandler stepExecutionRequestHandler;
+
+    @Autowired
+    private InputChannelInterceptor inputInterceptor;
+
+    @Autowired
+    private FineractProperties fineractProperties;
+
+    @Bean
+    public ConsumerFactory<Integer, Object> consumerFactory() {
+        FineractRemoteJobMessageHandlerKafkaProperties kafkaProperties = fineractProperties.getRemoteJobMessageHandler().getKafka();
+        Map<String, Object> props = new HashMap<>(kafkaProperties.getConsumer().getExtraPropertiesMap());
+        props.put(BOOTSTRAP_SERVERS_CONFIG, kafkaProperties.getBootstrapServers());
+        props.put(GROUP_ID_CONFIG, kafkaProperties.getConsumer().getGroupId());
+        props.put(KEY_DESERIALIZER_CLASS_CONFIG, IntegerDeserializer.class);
+        props.put(VALUE_DESERIALIZER_CLASS_CONFIG, JsonDeserializer.class);
+        props.put(TRUSTED_PACKAGES, "*");
+        props.put(AUTO_OFFSET_RESET_CONFIG, "earliest");
+        return new DefaultKafkaConsumerFactory<>(props);
+    }
+
+    @Bean
+    public ConcurrentKafkaListenerContainerFactory<Integer, Object> kafkaListenerContainerFactory(
+            ConsumerFactory<Integer, Object> consumerFactory) {
+        ConcurrentKafkaListenerContainerFactory<Integer, Object> factory = new ConcurrentKafkaListenerContainerFactory<>();
+        factory.setConsumerFactory(consumerFactory);
+        factory.getContainerProperties().setAckMode(ContainerProperties.AckMode.MANUAL);
+        return factory;
+    }
+
+    @KafkaListener(topics = "${fineract.remote-job-message-handler.kafka.topic.name}")
+    public void onMessage(@Payload ContextualMessage contextualMessage, Acknowledgment acknowledgment) {
```

Comment:
Probably I'd put this into a separate class.

Reply:
done